airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DefaultTyperDeduper.java

```diff
@@ -101,12 +104,25 @@ public DefaultTyperDeduper(
     this(sqlGenerator, destinationHandler, parsedCatalog, v1V2Migrator, new NoopV2TableMigrator(), defaultThreadCount);
   }
 
+  private void prepareSchemas(ParsedCatalog parsedCatalog) throws Exception {
+    var rawSchema = parsedCatalog.streams().stream().map(stream -> stream.id().rawNamespace());
+    var finalSchema = parsedCatalog.streams().stream().map(stream -> stream.id().finalNamespace());
+    var allSchemas = Streams.concat(rawSchema, finalSchema)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toSet());
+    if (allSchemas.size() > 2) {
```

Comment:
nit: this can be normal, the catalog can have streams from multiple namespaces (e.g. `public1.users`, `public2.users`)